### PR TITLE
Disable features for minors

### PR DESCRIPTION
**Pull request by the-kenneth**
The YourWork screen has been disabled for minors and the last 4 questions of AboutYou also get hidden if the entered age is a minor.

### CODE REVIEW

File: src/core/patient/PatientService.ts [hunk 1]
```diff
@@ -165,6 +164,8 @@ export class PatientService extends ApiClientBase implements IPatientService {
     const isNHSStudy = patient.is_in_uk_nhs_asymptomatic_study;
     const hasSchoolGroup = patient.has_school_group;
 
+    const isMinor = isMinorAge(patient.year_of_birth);
```
Comment (selvach) [inline src/core/patient/PatientService.ts:167]: Could this be null / non-numeric?

File: src/features/patient/AboutYouScreen.tsx [hunk 2]
```diff
@@ -127,6 +128,7 @@ export default class AboutYouScreen extends Component<AboutYouProps, State> {
           currentPatient.isFemale = formData.sex !== 'male';
           currentPatient.isPeriodCapable =
             !['', 'male', 'pfnts'].includes(formData.sex) || !['', 'male', 'pfnts'].includes(formData.genderIdentity);
+          currentPatient.isMinor = isMinorAge(cleanIntegerVal(formData.yearOfBirth));
```
Comment (selvach) [inline src/features/patient/AboutYouScreen.tsx:131]: Just checking if cleanIntegerVal needs a null check?

File: src/features/patient/AboutYouScreen.tsx [hunk 3]
```diff
@@ -428,29 +432,33 @@ export default class AboutYouScreen extends Component<AboutYouProps, State> {
...
-                    selectedValue={props.values.helpAvailable}
-                    onValueChange={props.handleChange('helpAvailable')}
-                  />
-
-                  <YesNoField
-                    label={i18n.t('mobility-aid')}
-                    selectedValue={props.values.mobilityAid}
-                    onValueChange={props.handleChange('mobilityAid')}
-                  />
+                  {!isMinor && (
```
Comment (selvach) [inline src/features/patient/AboutYouScreen.tsx:435]: Payload submitted to the backend contains `false` for these values:

```
    "housebound_problems": false,
    "needs_help": false,
    "help_available": false,
    "mobility_aid": false,
```

Which is better: `null` or assume `false`?

File: src/features/patient/AboutYouScreen.tsx (hunk 2 above)
Comment (the-kenneth) [reply src/features/patient/AboutYouScreen.tsx:131]: The form validation would not allow a submit if this is not correctly filled, thus you couldn't hit this code path.

File: src/features/patient/AboutYouScreen.tsx (hunk 3 above)
Comment (the-kenneth) [reply src/features/patient/AboutYouScreen.tsx:435]: My assumption was set as false, this means the current data set wouldnt have a new value introduced. I can check double check this.

File: src/core/patient/PatientService.ts (hunk 1 above)
Comment (the-kenneth) [reply src/core/patient/PatientService.ts:167]: This should never be null or non-numeric as a submit to the backend is validated previously. Am I correct in thinking that?

File: src/features/patient/AboutYouScreen.tsx (hunk 3 above)
Comment (the-kenneth) [reply src/features/patient/AboutYouScreen.tsx:441]: Ive made changes to not include those questions into the payload if not answered.

**Review by selvach — APPROVED** [review]

**Pull request merged**